src/wrappedGraphs/graphviews.jl

```diff
@@ -0,0 +1,125 @@
+"""
+    ReverseView{T<:Integer,G<:AbstractGraph} <: AbstractGraph{T}
+
+A graph view that wraps a directed graph and reverse the direction of every edge.
+
+# Examples
+```jldoctest
+julia> using Graphs
+
+julia> g = SimpleDiGraph(2);
+
+julia> add_edge!(g, 1, 2);
+
+julia> rg = ReverseView(g);
+
+julia> neighbors(rg, 1)
+Int64[]
+
+julia> neighbors(rg, 2)
+1-element Vector{Int64}:
+ 1
+```
+"""
+struct ReverseView{T<:Integer,G<:AbstractGraph} <: AbstractGraph{T}
+    g::G
+
+    @traitfn ReverseView{T,G}(g::::(IsDirected)) where {T<:Integer,G<:AbstractGraph{T}} =
+        new(g)
+    @traitfn ReverseView{T,G}(g::::(!IsDirected)) where {T<:Integer,G<:AbstractGraph{T}} =
+        throw(ArgumentError("Your graph needs to be directed"))
+end
```

Comment:
Wouldn't it make more sense if this function works on all graphs? Even if it does nothing for undirected graphs.